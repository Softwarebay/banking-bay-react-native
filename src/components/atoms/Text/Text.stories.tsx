import { text } from '@storybook/addon-knobs';
import { storiesOf } from '@storybook/react-native';
import { ViewStory } from 'utils/storybook/ViewStory';

import { Text } from './Text';

storiesOf('Atoms/Text', module)
	.addDecorator((getStory) => <ViewStory center>{getStory()}</ViewStory>)
	.add('Default', () => <Text text={text('text', 'Body text')} />)
	.add('Long text', () => (
		<Text
			text={text(
				'text',
				'Lorem ipsum dolor sit amet. Ex quidem iure vitae neque et expedita officia a laboriosam reprehenderit eos inventore doloremque sit voluptate aperiam et architecto iusto?'
			)}
		/>
	));
