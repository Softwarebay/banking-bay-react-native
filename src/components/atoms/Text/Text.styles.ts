import { StyleSheet } from 'react-native';

import { colors } from 'theme';

export const styles = StyleSheet.create({
	text: {
		color: colors.bodyText,
		fontFamily: 'MulishRegular',
		fontSize: 16,
		lineHeight: 25.6,
	},
});
