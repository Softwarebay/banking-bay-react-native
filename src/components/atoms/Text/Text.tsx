import { Text as RNText } from 'react-native';

import { styles } from './Text.styles';

interface Props {
	text: string;
	testID?: string;
}

export const Text = ({ text, testID }: Props) => {
	return (
		<RNText style={styles.text} testID={testID}>
			{text}
		</RNText>
	);
};
