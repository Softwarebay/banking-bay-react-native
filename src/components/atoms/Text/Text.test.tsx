import { render, screen } from '@testing-library/react-native';
import testRenderer from 'react-test-renderer';

import { Text } from './Text';

import type { Props } from './Text';

describe('Text', () => {
	const props: Props = {
		text: 'Text',
		style: {
			marginBottom: 20,
		},
		testID: 'text-test-id',
	};

	it('should match snapshot', () => {
		const tree = testRenderer.create(<Text {...props} />).toJSON();

		expect(tree).toMatchSnapshot();
	});

	it('should render text passed via props', () => {
		render(<Text {...props} />);

		const text = screen.getByTestId('text-test-id');

		expect(text).toHaveTextContent('Text');
	});

	it('should apply additional styles from props', () => {
		render(<Text {...props} />);

		const text = screen.getByTestId('text-test-id');

		expect(text).toHaveStyle({
			marginBottom: 20,
		});
	});
});
