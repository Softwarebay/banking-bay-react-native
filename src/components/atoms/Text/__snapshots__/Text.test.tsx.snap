// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Text should match snapshot 1`] = `
<Text
  style={
    Array [
      Object {
        "color": "#4C4C60",
        "fontFamily": "MulishRegular",
        "fontSize": 17,
        "lineHeight": 27.2,
      },
      Object {
        "marginBottom": 20,
      },
    ]
  }
  testID="text-test-id"
>
  Text
</Text>
`;
